src/bootstrap.coffee

```diff
@@ -32,20 +31,21 @@ bootstrap = ->
 		userConfig.deviceType ?= 'raspberry-pi'
 		if userConfig.registered_at?
 			return userConfig
-		deviceRegister.register(resinApi, userConfig)
-		.catch DuplicateUuidError, ->
-			resinApi.get
-				resource: 'device'
-				options:
-					filter:
-						uuid: userConfig.uuid
-				customOptions:
-					apikey: userConfig.apiKey
-			.then ([ device ]) ->
-				return device
-		.then (device) ->
+
+		deviceRegister.register(
+			userId: userConfig.userId
+			applicationId: userConfig.applicationId
+			uuid: userConfig.uuid
+			deviceType: userConfig.deviceType
+			deviceApiKey: userConfig.deviceApiKey
+			provisioningApiKey: userConfig.apiKey
+			apiEndpoint: config.apiEndpoint
+		)
+		.then ({ id }) ->
 			userConfig.registered_at = Date.now()
-			userConfig.deviceId = device.id
+			userConfig.deviceId = id
+			# Delete the provisioning key now.
+			delete userConfig.apiKey
```

Comment:
For dependent devices, we'll probably need to keep the provisioning key, unless the device key works to provision (and query) dependent devices.
See: https://github.com/resin-io/resin-supervisor/pull/238/files#diff-1cad6517feeffa5e7c8b889dc57f761aR35